test/integration/apiserver/apiserver_test.go

```diff
@@ -238,3 +238,79 @@ func TestAPIListChunking(t *testing.T) {
 		t.Errorf("unexpected items: %#v", list)
 	}
 }
+
+func TestNameInFieldSelector(t *testing.T) {
+	s, clientSet, closeFn := setup(t)
+	defer closeFn()
+
+	numNamespaces := 3
+	namespaces := make([]*v1.Namespace, 0, numNamespaces)
+	for i := 0; i < 3; i++ {
+		ns := framework.CreateTestingNamespace(fmt.Sprintf("ns%d", i), s, t)
+		defer framework.DeleteTestingNamespace(ns, s, t)
+		namespaces = append(namespaces, ns)
+
+		secret := &v1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "foo",
+			},
+			Data: map[string][]byte{
+				"key": []byte("value"),
+			},
+		}
+		_, err := clientSet.CoreV1().Secrets(ns.Name).Create(secret)
+		if err != nil {
+			t.Errorf("Couldn't create secret: %v", err)
+		}
```

Comment:
create a second secret `bar` in each namespace to make sure name filtering is happening

Reply:
done